app/screens/home/search/results/results.tsx

```diff
@@ -65,207 +60,57 @@ const SearchResults = ({
-                />
-            );
+    const styles = getStyles(dimensions);
+
+    const transform = useAnimatedStyle(() => {
+        const translateX = selectedTab === TabTypes.MESSAGES ? 0 : -dimensions.width;
+        return {
+            transform: [
+                {translateX: withTiming(translateX, {duration})},
+            ],
         };
-        bottomSheet({
-            closeButtonId: 'close-search-file-options',
-            renderContent,
-            snapPoints,
-            theme,
-            title: '',
-        });
-    }, [orderedFilesForGallery, snapPoints, theme]);
-
-    // This effect handles the case where a user has the FileOptions Modal
-    // open and the server changes the ability to download files or copy public
-    // links. Reopen the Bottom Sheet again so the new options are added or
-    // removed.
-    useEffect(() => {
-        if (lastViewedIndex === undefined) {
-            return;
-        }
-        if (NavigationStore.getNavigationTopComponentId() === 'BottomSheet') {
-            dismissBottomSheet().then(() => {
-                handleOptionsPress(lastViewedIndex);
-            });
-        }
-    }, [canDownloadFiles, publicLinkEnabled]);
+    }, [selectedTab, dimensions.width]);
 
-    const renderItem = useCallback(({item}: ListRenderItemInfo<string|FileInfo | Post>) => {
-        if (item === 'loading') {
-            return (
+    return (
+        <>
+            {loading &&
                 <Loading
                     color={theme.buttonBg}
                     size='large'
+                    containerStyle={[styles.loading, {paddingTop: scrollPaddingTop}]}
                 />
-            );
-        }
-
-        if (typeof item === 'string') {
-            if (isDateLine(item)) {
-                return (
-                    <DateSeparator
-                        date={getDateForDateLine(item)}
-                        timezone={isTimezoneEnabled ? currentTimezone : null}
+            }
+            {!loading &&
+            <Animated.View style={[styles.container, transform]}>
+                <View style={styles.result} >
+                    <PostResults
+                        currentTimezone={currentTimezone}
+                        isTimezoneEnabled={isTimezoneEnabled}
+                        posts={posts}
+                        scrollPaddingTop={scrollPaddingTop}
+                        searchValue={searchValue}
                     />
-                );
+                </View>
+                <View style={styles.result} >
+                    <FileResults
+                        canDownloadFiles={canDownloadFiles}
+                        fileChannels={fileChannels}
+                        fileInfos={fileInfos}
+                        publicLinkEnabled={publicLinkEnabled}
+                        scrollPaddingTop={scrollPaddingTop}
+                        searchValue={searchValue}
+                    />
+                </View>
```

Comment:
Just to make sure. We have mounted both components all the time, and we are only moving the parent view to the left or the right, to show the one we want to show. Am I right?

Reply:
That is correct.  That is was the biggest gain in speed. THey trouble being that rendering the posts with every change from files to posts was really slow.  Credit to @enahum for the suggestion.